mesa/visualization/components/matplotlib.py

```diff
@@ -23,12 +25,39 @@ def SpaceMatplotlib(model, agent_portrayal, dependencies: list[any] | None = Non
     solara.FigureMatplotlib(space_fig, format="png", dependencies=dependencies)
 
 
+# matplotlib scatter does not allow for multiple shapes in one call
+def _split_and_scatter(portray_data, space_ax):
+    grouped_data = defaultdict(lambda: {"x": [], "y": [], "s": [], "c": []})
+
+    # Extract data from the dictionary
+    x = portray_data["x"]
+    y = portray_data["y"]
+    s = portray_data["s"]
+    c = portray_data["c"]
+    m = portray_data["m"]
+
+    assert len(x) == len(y) == len(s) == len(c) == len(m)
```

Comment:
Can't use assertion here. Needs to raise an exception because this would be disabled by the Python `-O` flag (to disable assertions).